### PR TITLE
[REALM-360] Add testing utilities

### PR DESCRIPTION
**Pull request by drgmr**
Library rewrite:

- Adapter based interfacing
- Hides both Kaffe and Exq from clients
- Has a general purpose test adapter

Current issues:

- An `exq` configuration is necessary on clients when using `exq_ui` due to complications on how the app tree is built

### CODE REVIEW

**Comment by execb5** [comment]
For some reason @drgmr 's picture is not showing in 54bbc63

**Comment by drgmr** [comment]
@execb5 Thanks for pointing that out, something went wrong when migrating machines. It's fixed - also I've given credit to @machadogab.

File: lib/kafka_message_bus/adapters/kaffe.ex [hunk 1]
```diff
@@ -0,0 +1,77 @@
+defmodule KafkaMessageBus.Adapters.Kaffe do
+  alias Kaffe.Producer
+
+  alias KafkaMessageBus.Config
+  alias KafkaMessageBus.Adapters.Kaffe.Consumer
+
+  @behaviour KafkaMessageBus.Adapter
+
+  @impl true
```
Comment (Heckfer) [inline lib/kafka_message_bus/adapters/kaffe.ex]: replace with @impl ModuleName

File: lib/kafka_message_bus/adapters/kaffe.ex [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  def init(config) do
+    config
+    |> to_kaffe_config()
+    |> apply_kaffe_config()
+
+    start_kaffe()
+  end
+
+  @impl true
```
Comment (Heckfer) [inline lib/kafka_message_bus/adapters/kaffe.ex]: replace with @impl ModuleName

File: lib/kafka_message_bus/adapters/kaffe/consumer.ex [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+  defp run_consumers({:ok, contents}, topic) do
+    case get_consumers_for(topic, contents["resource"]) do
+      [] ->
+        Logger.error(fn ->
+          "No consumers configured for #{topic}/#{contents["resource"]}"
+        end)
+
+      consumers ->
+        Enum.each(consumers, fn consumer -> run_consumer(consumer, contents) end)
```
Comment (Heckfer) [inline lib/kafka_message_bus/adapters/kaffe/consumer.ex]: add log on message accepted

Comment (Heckfer) [reply lib/kafka_message_bus/adapters/kaffe/consumer.ex]: log topic, resource, id?

Comment (Heckfer) [reply lib/kafka_message_bus/adapters/kaffe/consumer.ex]: make it a `info` log

File: lib/kafka_message_bus/adapters/kaffe/consumer.ex [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    message.value
+    |> Poison.decode()
+    |> run_consumers(message.topic)
+  end
+
+  defp run_consumers({:ok, contents}, topic) do
+    case get_consumers_for(topic, contents["resource"]) do
+      [] ->
+        Logger.error(fn ->
+          "No consumers configured for #{topic}/#{contents["resource"]}"
```
Comment (Heckfer) [inline lib/kafka_message_bus/adapters/kaffe/consumer.ex]: log the id as well?

**Pull request merged**